### ISSUE TITLE
adb shell does not provides root user, instead shell user is provided

### ISSUE THREAD

**Issue by serdeliuk** [TeamWin/Team-Win-Recovery-Project#1639]
- Custom built, porting to a new  device
- [ *] I am running the latest version of TWRP
- [ *] I have read the FAQ (https://twrp.me/FAQ/)
- [ *] I have searched for my issue and it does not already exist

**Device codename**: q5q SM-F946B
**TWRP version**: 3.7.0

#### WHAT STEPS WILL REPRODUCE THE PROBLEM?
~$ adb shell
q5q:/ # whoami                                                                                                                                                                                                    
shell
q5q:/ # 

#### WHAT IS THE EXPECTED RESULT?
the user should be root not shell

#### WHAT HAPPENS INSTEAD?
running from recovery image

#### ADDITIONAL INFORMATION
N/A

**Comment by CaptainThrowback** [comment]
You built as -eng, not -userdebug, right?

**Comment by serdeliuk** [comment]
yes, eng, I am using prebuilt kernel from stock recovery, if that counts

**Comment by serdeliuk** [comment]
and creating recovery.img

**Comment by serdeliuk** [comment]
what is interesting is that from the  twrp terminal i have root
![image](https://github.com/TeamWin/Team-Win-Recovery-Project/assets/38280618/6724a45c-2300-4f3c-a79a-1f6dda48f1c9)

**Comment by CaptainThrowback** [comment]
> what is interesting is that from the  twrp terminal i have root
> ![image](https://github.com/TeamWin/Team-Win-Recovery-Project/assets/38280618/6724a45c-2300-4f3c-a79a-1f6dda48f1c9)
> 

You sure this issue isn't caused by something in your device tree?

**Comment by serdeliuk** [comment]
for sure is something in my device tree, but what, this is what i am asking for, what could generate this behaviour?

**Comment by CaptainThrowback** [comment]
> for sure is something in my device tree, but what, this is what i am asking for, what could generate this behaviour?

Did I miss the link to your device tree?

**Comment by serdeliuk** [comment]
indeed, this is https://github.com/serdeliuk/android_device_samsung_dm3q i only modified the device name

**Comment by CaptainThrowback** [comment]
> indeed, this is https://github.com/serdeliuk/android_device_samsung_dm3q i only modified the device name

Nothing jumps out at me, sorry.

**Comment by serdeliuk** [comment]
It seems to be something in the kernel, not sure what, but for sure is a config that needs to be disabled
